src/features/applications/hooks/defaults.ts

```diff
@@ -14,6 +15,30 @@ hooks.addPureHook('POST', 'resin', 'application', {
 			);
 		}
 
+		request.values.uuid = request.values.uuid ?? uuid.v4().replace(/-/g, '');
+
+		// reconstitute the value into a properly formatted UUID...
+		const toUuid = (i: string): string => {
```

Comment:
We probably also don't need to redefine this function each time that the hook runs.